File: server/garant/src/main/java/com/thedeveloper/garant/entity/DealEntity.java
package com.thedeveloper.garant.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;

@Entity
@Table(name = "deals")
@Getter
@Setter
public class DealEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    String id;

    String name;
    String description;
    int duration;
    double price;
    Date startDate;
    Date endDate;
    String type;
    String file;
    String code;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "seller", referencedColumnName = "phone")
    UserEntity seller;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "buyer", referencedColumnName = "phone")
    UserEntity buyer;
}

File: app/lib/api/entity/DealEntity.dart
import 'package:app/api/entity/UserEntity.dart';
import 'package:json_annotation/json_annotation.dart';

part 'DealEntity.g.dart';

@JsonSerializable()
class DealEntity{
  String id;
  String name;
  String description;
  int duration;
  double price;
  String type;
  DateTime startDate;
  DateTime? endDate;
  String? file;
  String code;
  UserEntity? seller;
  UserEntity? buyer;

  DealEntity({
    required this.id,
    required this.name,
    required this.description,
    required this.duration,
    required this.price,
    required this.type,
    required this.startDate,
    this.endDate,
    this.file,
    required this.code,
    this.seller,
    this.buyer
});

  factory DealEntity.fromJson(Map<String, dynamic> json) => _$DealEntityFromJson(json);
  Map<String, dynamic> toJson() => _$DealEntityToJson(this);
}